evdns.c

```diff
@@ -4631,48 +4631,60 @@ load_nameservers_with_getnetworkparams(struct evdns_base *base)
 		status = -1;
 		goto done;
 	}
-	if (!(fn = (GetNetworkParams_fn_t) GetProcAddress(handle, "GetNetworkParams"))) {
+	if (!(fn = (GetAdaptersAddresses_fn_t) GetProcAddress(handle, "GetAdaptersAddresses"))) {
 		log(EVDNS_LOG_WARN, "Could not get address of function.");
 		status = -1;
 		goto done;
 	}
 
 	buf = mm_malloc(size);
 	if (!buf) { status = 4; goto done; }
-	fixed = buf;
-	r = fn(fixed, &size);
-	if (r != ERROR_SUCCESS && r != ERROR_BUFFER_OVERFLOW) {
+	addresses = buf;
+	r = fn(AF_UNSPEC, GAA_FLAG_INCLUDE_PREFIX, NULL, addresses, &size);
+	if (r != NO_ERROR && r != ERROR_BUFFER_OVERFLOW) {
 		status = -1;
 		goto done;
 	}
-	if (r != ERROR_SUCCESS) {
+	if (r != NO_ERROR) {
 		mm_free(buf);
 		buf = mm_malloc(size);
 		if (!buf) { status = 4; goto done; }
-		fixed = buf;
-		r = fn(fixed, &size);
-		if (r != ERROR_SUCCESS) {
+		addresses = buf;
+		r = fn(AF_UNSPEC, GAA_FLAG_INCLUDE_PREFIX, NULL, addresses, &size);
+		if (r != NO_ERROR) {
 			log(EVDNS_LOG_DEBUG, "fn() failed.");
 			status = -1;
 			goto done;
 		}
 	}
+	EVUTIL_ASSERT(addresses);
 
-	EVUTIL_ASSERT(fixed);
-	added_any = 0;
-	ns = &(fixed->DnsServerList);
-	while (ns) {
-		r = evdns_nameserver_ip_add_line(base, ns->IpAddress.String);
-		if (r) {
-			log(EVDNS_LOG_DEBUG,"Could not add nameserver %s to list,error: %d",
-				(ns->IpAddress.String),(int)GetLastError());
-			status = r;
-		} else {
-			++added_any;
-			log(EVDNS_LOG_DEBUG,"Successfully added %s as nameserver",ns->IpAddress.String);
-		}
+	while (addresses) {
+		dnserver = addresses->FirstDnsServerAddress;
+		while (dnserver && (addresses->OperStatus == IfOperStatusUp)) {
+			char ip[INET6_ADDRSTRLEN] = {0};
+			if (AF_INET == dnserver->Address.lpSockaddr->sa_family) {
+				inet_ntop(AF_INET, &((SOCKADDR_IN *)dnserver->Address.lpSockaddr)->sin_addr, ip, sizeof(ip));
+			} else if (AF_INET6 == dnserver->Address.lpSockaddr->sa_family) {
+				inet_ntop(AF_INET6, &((SOCKADDR_IN6 *)dnserver->Address.lpSockaddr)->sin6_addr, ip, sizeof(ip));
+			}
 
-		ns = ns->Next;
+			dnserver = dnserver->Next;
+			if (strncmp(ip, "fec0:", 5) == 0) { /* remove ipv6 reserved address */
+				continue;
+			}
+			
+			r = evdns_base_nameserver_ip_add(base, ip);
+			if (r) {
+				log(EVDNS_LOG_DEBUG,"Could not add nameserver %s to list,error: %d", (ip), r);
```

Comment:
```suggestion
				log(EVDNS_LOG_DEBUG, "Could not add nameserver %s to list, error: %d", ip, r);
```

Also, can we get string value of the error?